File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/60968-Visual Studio 2010 SDK Samples/VSSDK IDE Sample Menu And Commands/VB/TDD/TestVsPkg.vb
'**************************************************************************

'Copyright (c) Microsoft Corporation. All rights reserved.
'This code is licensed under the Visual Studio SDK license terms.
'THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
'ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
'IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
'PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

'**************************************************************************

' System references.

Imports Microsoft.VisualBasic
Imports System
Imports System.Reflection

' Platform references.
Imports Microsoft.VisualStudio.Shell.Interop

' Unit Test framework.
Imports Microsoft.VsSDK.UnitTestLibrary
Imports Microsoft.VisualStudio.TestTools.UnitTesting

' Assembly to test.
Imports Microsoft.Samples.VisualStudio.MenuCommands

Namespace Microsoft.Samples.VisualStudio.MenuCommands.UnitTest

    <TestClass()> _
    Public Class MenuAndCommandPackageTest

        <TestMethod()> _
        Public Sub PackageCreation()
            Dim package As New MenuCommandsPackage()
            Assert.IsNotNull(package, "Failed to create package")
        End Sub

        <TestMethod()> _
        Public Sub PackageInitialize()
            Dim packageObject As New MenuCommandsPackage()
            Assert.IsNotNull(packageObject, "Failed to create package")
            Dim package As IVsPackage = TryCast(packageObject, IVsPackage)
            Assert.IsNotNull(package, "Can not get the package interface from the package object.")
            Using serviceProvider As OleServiceProvider = OleServiceProvider.CreateOleServiceProviderWithBasicServices()
                Assert.AreEqual(0, package.SetSite(serviceProvider), "Can not get the package interface from the package object.")
            End Using
        End Sub

        <TestMethod()> _
        Public Sub PackageInitializeWithNull()
            Dim packageObject As New MenuCommandsPackage()
            Assert.IsNotNull(packageObject, "Failed to create package")
            Dim package As IVsPackage = TryCast(packageObject, IVsPackage)
            Assert.IsNotNull(package, "Can not get the package interface from the package object.")
            Assert.AreEqual(0, package.SetSite(Nothing), "SetSite returned an unexpected value")
        End Sub
    End Class

End Namespace

File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/60968-Visual Studio 2010 SDK Samples/VSSDK IDE Sample Menu And Commands/C#/TDD/TestVsPkg.cs
/***************************************************************************

Copyright (c) Microsoft Corporation. All rights reserved.
This code is licensed under the Visual Studio SDK license terms.
THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

***************************************************************************/

// System references.
using System;
using System.Reflection;

// Platform references
using Microsoft.VisualStudio.Shell.Interop;

// Unit Test framework
using Microsoft.VsSDK.UnitTestLibrary;
using Microsoft.VisualStudio.TestTools.UnitTesting;

// Assembly to test
using Microsoft.Samples.VisualStudio.MenuCommands;

namespace Microsoft.Samples.VisualStudio.MenuAndCommands.UnitTest {

    [TestClass()]
    public class MenuAndCommandPackageTest {

        [TestMethod()]
        public void PackageCreation() {
            MenuCommandsPackage package = new MenuCommandsPackage();
            Assert.IsNotNull(package, "Failed to create package");
        }

        [TestMethod()]
        public void PackageInitialize() {
            MenuCommandsPackage packageObject = new MenuCommandsPackage();
            Assert.IsNotNull(packageObject, "Failed to create package");
            IVsPackage package = packageObject as IVsPackage;
            Assert.IsNotNull(package, "Can not get the package interface from the package object.");
            using (OleServiceProvider serviceProvider = OleServiceProvider.CreateOleServiceProviderWithBasicServices())
            {
                Assert.AreEqual(0, package.SetSite(serviceProvider), "Can not get the package interface from the package object.");
            }
        }

        [TestMethod()]
        public void PackageInitializeWithNull()
        {
            MenuCommandsPackage packageObject = new MenuCommandsPackage();
            Assert.IsNotNull(packageObject, "Failed to create package");
            IVsPackage package = packageObject as IVsPackage;
            Assert.IsNotNull(package, "Can not get the package interface from the package object.");
            Assert.AreEqual(0, package.SetSite(null), "SetSite returned an unexpected value");
        }
    }

}
